### ISSUE TITLE
Online Co-op: Server not replying when client wants to load room?

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#820]
Although this has been tightened up greatly in recent patches... still getting a report that server is not replying when player is trying to enter a room.

v0.2.2a

Feedback ID:  20210103-RVQ29

```
Server: Lightning-Beijing-01
GameID: YQSMB
```

```
9:08:52, Frame 665, LOG »  RELAY CLIENT: Connecting to server...
9:08:53, Frame 687, LOG »  NETWORK CLIENT: Connecting to 123.56.71.67:5889 ...
9:08:53, Frame 690, LOG »  NETWORK CLIENT: RelayClient connected to server 123.56.71.67:5889

9:08:53, Frame 693, LOG »  NETWORK HANDSHAKE: Connected.
Device is online.
Server: 
GameID: YQSMB


9:08:54, Frame 714, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPlayerHasFadedToBlack:
Sending message to server...


9:08:54, Frame 724, LOG »  NetworkMessage_DungeonRoomTransitionHelper:
Received message.
peekDesire: 4

9:08:55, Frame 746, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第16个房间：蓝区

9:12:44, Frame 9446, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPartyShouldStartUsingCorridor:
Sending message to server...

[[[ ^ REPEATING MESSAGE... ]]]

9:13:21, Frame 10835, LOG »  RELAY CLIENT: Connecting to server...
9:13:22, Frame 10853, LOG »  NETWORK CLIENT: Connecting to 123.56.71.67:5889 ...
9:13:22, Frame 10854, LOG »  NETWORK CLIENT: RelayClient connected to server 123.56.71.67:5889

9:13:22, Frame 10856, LOG »  NETWORK HANDSHAKE: Connected.
Device is online.
Server: Lightning-Beijing-01
GameID: YQSMB


9:13:23, Frame 10875, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPlayerHasFadedToBlack:
Sending message to server...


9:13:23, Frame 10878, LOG »  NetworkMessage_DungeonRoomTransitionHelper:
Received message.
peekDesire: 4

9:13:24, Frame 10900, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第16个房间：蓝区

9:14:24, Frame 13659, LOG »  NetworkMessage_DungeonRoomTransitionHelper.TellServerPartyShouldStartUsingWaypoint:
Sending message to server...

[[[ ^ REPEATING MESSAGE... ]]]
```

**Comment by ChrisNZL** [comment]
### Additional report
0.2.2a
Feedback ID: 20210117-WQDLV
Server: HostUS-HongKong-01
GameID: BCRXF

> If you enter another room during coop, the game wil l go dark and stop responding. if i quit the game and start it again, I can access the sever, but I cant move from the room.

**Comment by ChrisNZL** [comment]
Fixed in 0.2.3:

- Room transitions no longer stall for all clients if one client happens to fail loading or disconnects mid-transition.

- Fixed player creatures not having their world position be updated properly when changing rooms. Previously, this was not noticed client-side when loading properly, but world positions were not getting set server-side until the client reported their new position after the room loaded, which would cause issues if they disconnected mid-load.
